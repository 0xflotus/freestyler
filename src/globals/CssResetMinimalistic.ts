import global from './global';

const css = {
    '*': {
        pad: 0,
        mar: 0,
    },
};

const CssResetMinimalistic = global(css);

export default CssResetMinimalistic;
