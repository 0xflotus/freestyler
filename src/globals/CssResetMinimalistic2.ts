import {global} from './index';

export default global({
    '*': {
        pad: 0,
        mar: 0,
        bd: 0,
    },
});
