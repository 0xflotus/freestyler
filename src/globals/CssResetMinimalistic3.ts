import globalCss from './global';

const css = {
    '*': {
        pad: 0,
        mar: 0,
        bd: 0,
        out: 0,
    },
};

const CssResetMinimalistic3 = globalCss(css);

export default CssResetMinimalistic3;
