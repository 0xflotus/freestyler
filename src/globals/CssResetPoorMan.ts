import global from './global';

const css = {
    'html, body': {
        pad: 0,
        mar: 0,
    },
    html: {
        fz: '1em',
    },
    body: {
        fz: '100%',
    },
    'a img, :link img, :visited img': {
        bd: 0,
    },
};

const CssResetPoorMan = global(css);

export default CssResetPoorMan;
