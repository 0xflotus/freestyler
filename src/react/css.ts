import {Component, cloneElement} from 'react';
import {TCssTemplate} from '../types/index';
import decoratorCssComponent from './decorator/cssComponent';
import decoratorClass from './decorator/class';
import decoratorMethod from './decorator/method';
import {styleit} from './styleit';

export type TCssDecorator = any;

const isReactComponent = f => !!(f && f.prototype && f.prototype.render);

const css: TCssDecorator = (a?: TCssTemplate | any) => {
    // If component decorator without arguments.
    if (isReactComponent(a)) {
        decoratorCssComponent(a);
        return;
    }

    return (instanceOrComp, key, descriptor) =>
        typeof key === 'string'
            ? decoratorMethod(a)(instanceOrComp, key, descriptor)
            : decoratorClass(a)(instanceOrComp);
};

export default css;
