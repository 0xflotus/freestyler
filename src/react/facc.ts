import {TCssTemplate} from '../types/index';
import {TComponentTag} from '../types/base';
import wrap from './wrap';

function facc(Element: TComponentTag<any> = 'div', template: TCssTemplate = null) {
    let dynamicTemplate = null;
    const Comp = wrap(Element, template, () => dynamicTemplate, 'facc');

    return childrenCallback => {
        return (...args) => {
            let ast;
            [dynamicTemplate, ast] = childrenCallback(...args)(Comp);
            return ast;
        };
    };
}

export default facc;
