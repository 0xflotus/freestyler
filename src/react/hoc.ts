import {TCssTemplate, TComponentTag, TComponentType, TCssTemplateCallback} from '../types/index';
import wrap from './wrap';

const hoc = (staticTemplate?, dynamicTemplate?) => type => wrap(type, staticTemplate, dynamicTemplate);

export default hoc;
