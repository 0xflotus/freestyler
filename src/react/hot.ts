import hot from './hot/implementation0';

export default hot;
