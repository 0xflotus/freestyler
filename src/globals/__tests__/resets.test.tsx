import {createElement as h} from 'react';
import CssResetEricMeyer from '../CssResetEricMeyer';
import CssResetEricMeyerCondensed from '../CssResetEricMeyerCondensed';
import CssResetMinimalistic from '../CssResetMinimalistic';
import CssResetMinimalistic2 from '../CssResetMinimalistic2';
import CssResetMinimalistic3 from '../CssResetMinimalistic3';
import CssResetPoorMan from '../CssResetPoorMan';
import CssResetShaunInman from '../CssResetShaunInman';
import CssResetSiolon from '../CssResetSiolon';
import CssResetTantek from '../CssResetTantek';
import CssResetTripoli from '../CssResetTripoli';
import {mount} from 'enzyme';

describe('globals', () => {
  describe('CSS resets', () => {
    it('<CssResetEricMeyer>', () => {
        const wrapper = mount(<CssResetEricMeyer />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetEricMeyerCondensed>', () => {
        const wrapper = mount(<CssResetEricMeyerCondensed />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetMinimalistic>', () => {
        const wrapper = mount(<CssResetMinimalistic />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetMinimalistic2>', () => {
        const wrapper = mount(<CssResetMinimalistic2 />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetMinimalistic3>', () => {
        const wrapper = mount(<CssResetMinimalistic3 />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetPoorMan>', () => {
        const wrapper = mount(<CssResetPoorMan />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetShaunInman>', () => {
        const wrapper = mount(<CssResetShaunInman />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetSiolon>', () => {
        const wrapper = mount(<CssResetSiolon />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetTantek>', () => {
        const wrapper = mount(<CssResetTantek />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('<CssResetTripoli>', () => {
        const wrapper = mount(<CssResetTripoli />);

        expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
