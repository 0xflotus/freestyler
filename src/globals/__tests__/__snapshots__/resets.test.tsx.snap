// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globals CSS resets <CssResetEricMeyer> 1`] = `"<style>html,body,div,span,applet,object,iframe,table,caption,tbody,tfoot,thead,tr,th,td,del,dfn,em,font,img,ins,kbd,q,s,samp,small,strike,strong,sub,sup,tt,var,h1,h2,h3,h4,h5,h6,p,blockquote,pre,a,abbr,acronym,address,big,cite,code,dl,dt,dd,ol,ul,li,fieldset,form,label,legend{vertical-align:baseline;font-family:inherit;font-weight:inherit;font-style:inherit;font-size:100%;outline:0;padding:0;margin:0;border:0;}:focus{outline:0;}body{background:white;line-height:1;color:black;}ol, ul{list-style:none;}table{border-collapse:separate;border-spacing:0;}caption, th, td{font-weight:normal;text-align:left;}blockquote:before, blockquote:after, q:before, q:after{content:\\"\\";}blockquote, q{quotes:\\"\\" \\"\\";}</style>"`;

exports[`globals CSS resets <CssResetEricMeyerCondensed> 1`] = `"<style>body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,form,fieldset,input,textarea,p,blockquote,th,td{padding:0;margin:0;}fieldset, img{border:0;}table{border-collapse:collapse;border-spacing:0;}ol, ul{list-style:none;}address, caption, cite, code, dfn, em, strong, th, var{font-weight:normal;font-style:normal;}caption, th{text-align:left;}h1, h2, h3, h4, h5, h6{font-weight:normal;font-style:100%;}q:before, q:after{content:'';}abbr, acronym{border:0;}</style>"`;

exports[`globals CSS resets <CssResetMinimalistic> 1`] = `"<style>*{padding:0;margin:0;}</style>"`;

exports[`globals CSS resets <CssResetMinimalistic2> 1`] = `"<style>*{padding:0;margin:0;border:0;}</style>"`;

exports[`globals CSS resets <CssResetMinimalistic3> 1`] = `"<style>*{padding:0;margin:0;border:0;outline:0;}</style>"`;

exports[`globals CSS resets <CssResetPoorMan> 1`] = `"<style>html, body{padding:0;margin:0;}html{font-size:1em;}body{font-size:100%;}a img, :link img, :visited img{border:0;}</style>"`;

exports[`globals CSS resets <CssResetShaunInman> 1`] = `"<style>body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,form,fieldset,input,p,blockquote,table,th,td,embed,object{padding:0;margin:0;}table{border-collapse:collapse;border-spacing:0;}fieldset,img,abbr{border:0;}address,caption,cite,code,dfn,em,h1,h2,h3,h4,h5,h6,strong,th,var{font-weight:normal;font-style:normal;}ul{list-style:none;}caption,th{text-align:left;}h1,h2,h3,h4,h5,h6{font-size:1.0em;}q:before,q:after{content:\\"\\";}a,ins{text-decoration:none;}</style>"`;

exports[`globals CSS resets <CssResetSiolon> 1`] = `"<style>*{vertical-align:baseline;font-family:inherit;font-style:inherit;font-size:100%;border:none;padding:0;margin:0;}body{padding:5px;}h1, h2, h3, h4, h5, h6, p, pre, blockquote, form, ul, ol, dl{margin:20px 0;}li, dd, blockquote{margin-left:40px;}table{border-collapse:collapse;border-spacing:0;}</style>"`;

exports[`globals CSS resets <CssResetTantek> 1`] = `"<style>:link,:visited{text-decoration:none;}ul,ol{list-style:none;}h1,h2,h3,h4,h5,h6,pre,code,p{font-size:1em;}ul,ol,dl,li,dt,dd,h1,h2,h3,h4,h5,h6,pre,form,body,html,p,blockquote,fieldset,input{padding:0;margin:0;}a img,:link img,:visited img{border:none;}address{font-style:normal;}</style>"`;

exports[`globals CSS resets <CssResetTripoli> 1`] = `"<style>*{text-decoration:none;font-size:1em;outline:none;padding:0;margin:0;}code,kbd,samp,pre,tt,var,textarea,input,select,isindex,listing,xmp,plaintext{white-space:normal;font-size:1em;font:inherit;}dfn,i,cite,var,address,em{font-style:normal;}th,b,strong,h1,h2,h3,h4,h5,h6{font-weight:normal;}a,img,a img,iframe,form,fieldset,abbr,acronym,object,applet,table{border:none;}table{border-collapse:collapse;border-spacing:0;}caption,th,td,center{vertical-align:top;text-align:left;}body{background:white;line-height:1;color:black;}q{quotes:\\"\\" \\"\\";}ul,ol,dir,menu{list-style:none;}sub,sup{vertical-align:baseline;}a{color:inherit;}hr{display:none;}font{color:inherit !important;font:inherit !important;}marquee{overflow:inherit !important;-moz-binding:none;}blink{text-decoration:none;}nobr{white-space:normal;}</style>"`;
